scripts/reporting/ncanda_quality_control_script.py

```diff
@@ -0,0 +1,315 @@
+    """
+    Generate a report indicating which Saliva Samples have not been entered.
+    """
+    # visit_ignore____yes with value 0 is not ignored
+    error = dict()
+    if math.isnan(row.get('exclude')):
+        if row.get('visit_ignore___yes') != 1:
+            # saliva_sample is not missing if saliva_sample_missing if value zero
+            if row.get('saliva_missing') != 1:
+                if row.get(saliva_collected) == 1:
+                # for form_date, date is stored as a string
+                    if type(row.get(saliva_date)) == float:
+                        error = dict(subject_site_id = idx[0],
+                                visit_date = row.get('visit_date'),
+                                event_name = idx[1],
+                                sample_missing = saliva_collected,
+                                visit_notes = row.get('visit_notes'),
+                                error = 'ERROR: Saliva Sample is missing'
+                                )
+    return error
+
+def visit_data_missing(idx,row):
+	"""
+	Generate a report indicating which Visit Dates are missing.
+	"""
+	error = dict()
+	if row.get('exclude') != 1:
+		if row.get('visit_ignore___yes') != 1:
+			if type(row.get('visit_date')) != str:
+				error = dict(subject_site_id = idx[0],
+					event_name = idx[1],
+					error = 'ERROR: Visit date missing.'
+					)
+	return error
+
+def WAIS_score_verification(idx,row):
+	"""
+	Verifies whether the wais_rawscore was computed correctly.
+	"""
+	# visit_ignore____yes with value 0 is not ignored
+	error = dict()
+	if math.isnan(row.get('exclude')):
+		if row.get('visit_ignore___yes') != 1:
+			# form is not missing if form_missing if value nan or zero
+			if row.get('np_wais4_missing') != 1:
+				if row.get('np_wais4_rawscore_computed') == row.get('np_wais4_rawscore_diff(correct)'):
+					if row.get('np_wais4_rawscore_diff(correct)') != 0:
+						error = dict(subject_site_id = idx[0],
+							visit_date = row.get('visit_date'),
+							event_name = idx[1],
+							error = 'ERROR: WAIS score is not verified'
+							)
+	return error
+
+def youth_report_sex(idx,row, field_missing, field_sex):
+	"""
+	Checks whether or not sex was entered correctly in the Youth Report
+	"""
+	# visit_ignore____yes with value 0 is not ignored
+	error = dict()
+	if math.isnan(row.get('exclude')):
+		if row.get('visit_ignore___yes') != 1:
+			# np is not missing if field_missing if value nan or zero
+			if row.get(field_missing) != 1:
+				if row.get('sex') != row.get(field_sex):
+					error = dict(subject_site_id = idx[0],
+								visit_date = row.get('visit_date'),
+								event_name = idx[1],
+								field = field_sex,
+								error = 'ERROR: SEX and SEX in YOUTHREPORT do not match.'
+								)
+	return error
+
+
+def main(args=None):
+    project_entry = get_project_entry()
+    project_df = data_entry_fields(fields,project_entry,'1y_visit_arm_1')
+    error = []
+
+    for idx, row in project_df.iterrows():
```

Comment:
@abonil91 since you pass the same parameters to almost all of these you could write a function to run these rather than using `if check` ... `if check:` ...